subprojects/ide-native/src/main/groovy/org/gradle/ide/visualstudio/plugins/VisualStudioPlugin.java

```diff
@@ -188,26 +202,30 @@ private void addTasksForVisualStudioProject(VisualStudioProjectInternal vsProjec
     }
 
     private void configureCleanTask() {
-        final Delete cleanTask = (Delete) getCleanTask();
+        final TaskProvider<Delete> cleanTask = (TaskProvider<Delete>) getCleanTask();
 
-        project.getTasks().withType(GenerateSolutionFileTask.class).all(new Action<GenerateSolutionFileTask>() {
+        cleanTask.configure(new Action<Delete>() {
             @Override
-            public void execute(GenerateSolutionFileTask task) {
-                cleanTask.delete(task.getOutputs().getFiles());
-            }
-        });
+            public void execute(Delete cleanTask) {
+                cleanTask.delete(new Callable<FileCollection>() {
+                    @Override
+                    public FileCollection call() throws Exception {
+                        ConfigurableFileCollection result = project.files();
+                        for (GenerateSolutionFileTask task : project.getTasks().withType(GenerateSolutionFileTask.class)) {
```

Comment:
There's 2 levels of laziness here (configuring the clean task and then deferring the lookup of the generator tasks). We only need one. If the clean task needs to be configured, it's fine to lookup the other tasks at that point and we can remove the inner laziness.

Reply:
Good point, changed.